ts/main.ts

```diff
@@ -1,64 +1,170 @@
-  entry.entryURL = formURL.value;
-  entry.entryTextArea = formTextArea.value;
-  entry.entryId = data.nextEntryId;
   data.nextEntryId = data.nextEntryId + 1;
-  // adding new post to te end which is wrong because new posts have to be at the top
-  // data.entries.push(entry);
-  // adding the new post data to the top, so we know its a new post
+
+  // every time we hit submit, a DOM tree is created or one entry li and appending it to ul a list
+  // of entries
+  // so its created quickly after submit which solves the issue of refreshing
+  // adding the new post data to the top or beginning, so we know its a new post
   data.entries.unshift(entry);
 
+  // adding only the newly created entry not the whole entries
+  // prepend adds the new li at the beginning not at the end which is similar to unshift()
+  // to maintain the same order in the DOM or html file like in the array entries
+  $ulList.prepend(renderEntry(entry));
 
-  writeEntry();
-  $img.src = "images/placeholder-image-square.jpg"
+  writeData();
+  $img.src = 'images/placeholder-image-square.jpg';
   $form.reset();
+  // when the form reset it immediately goes to entries
+  viewSwap('entries');
+  toggleNoEntries();
+});
+
+// from somewhere an entry will be passed to renderEntry function
+// returning an entry
+function renderEntry(entry: Entry): HTMLLIElement {
+  const $parentLI = document.createElement('li');
+  const $divRow = document.createElement('div');
+  $divRow.setAttribute('class', 'row');
+  // after being appended, it will be outputted because its added to the DOM
+  $parentLI.appendChild($divRow);
+  const $divColHalfFirst = document.createElement('div');
+  $divColHalfFirst.setAttribute('class', 'column-half');
+  $divRow.appendChild($divColHalfFirst);
+  const $img = document.createElement('img');
+  $img.setAttribute('class', 'entries-img');
+  $img.setAttribute('src', entry.entryURL);
+  $img.setAttribute('alt', 'mountain view');
+  $divColHalfFirst.appendChild($img);
+  const $divColHalfSecond = document.createElement('div');
+  $divColHalfSecond.setAttribute('class', 'column-half');
+  $divRow.appendChild($divColHalfSecond);
+  const $h2 = document.createElement('h2');
+  $h2.textContent = entry.entryTitle;
+  $divColHalfSecond.appendChild($h2);
+  const $h5 = document.createElement('h5');
+  $h5.textContent = entry.entryTextArea;
+  $divColHalfSecond.appendChild($h5);
+
+  return $parentLI;
+}
+
+// safety function because even if we don't create it it will load everything
+// all our elements in ul are created in our DOM or the page
+document.addEventListener('DOMContentLoaded', () => {
+  // creating a one entry and appending it to list ul
+
+  for (let i = 0; i < data.entries.length; i++) {
+    $ulList.appendChild(renderEntry(data.entries[i]));
+  }
+  // we are testing it  here because when our elements are created
+  // we call toggle to switch between adding h2 no entries or seeing the entries if they are exists
+  viewSwap('entry-form');
```

Comment:
You should be utilizing `data.view` here to swap to the last view the user was on